webnn/resources/test_data/reduce_mean.json

```diff
@@ -0,0 +1,782 @@
+            -3.3605690022427694,
+            -88.7479514780816
+          ],
+          "type": "float32"
+        }
+      },
+      "options": {
+        "axes": [-1]
+      },
+      "expected": {
+        "name": "output",
+        "shape": [2, 3],
+        "data": [
+          -5.550758361816406,
+          -29.72635269165039,
+          14.450223922729492,
+          9.080665588378906,
+          7.329422950744629,
+          -23.158693313598633
+        ],
+        "type": "float32"
+      }
+    },
+    {
+      "name": "reduceMean float32 4D tensor options.axes",
+      "inputs": {
+        "input": {
+          "shape": [2, 2, 2, 3],
+          "data": [
+            -1.913036787607325,
+            95.38300711992477,
+            -82.72358015843713,
+            -32.94943032326678,
+            91.42181191536278,
+            -75.1219094678544,
+            -70.81411336302938,
+            -64.39120554424065,
+            -46.982829731397715,
+            91.99561137130661,
+            26.951267233577482,
+            -14.163154887017143,
+            48.66138565461631,
+            -72.48218202176906,
+            -24.52633400273743,
+            84.66979710011137,
+            -27.139524955053318,
+            5.39985222686434,
+            48.46895687574735,
+            2.588408784898675,
+            15.988198777438896,
+            -16.514456459968002,
+            -3.3605690022427694,
+            -88.7479514780816
+          ],
+          "type": "float32"
+        }
+      },
+      "options": {
+        "axes": [0, 2]
+      },
+      "expected": {
+        "name": "output",
+        "shape": [2, 3],
+        "data": [
+          24.61717987060547,
+          21.795778274536133,
+          -44.24299240112305,
+          13.284000396728516,
+          -9.553024291992188,
+          -33.47643280029297
+        ],
+        "type": "float32"
+      }
+    },
+    {
+      "name": "reduceMean float32 4D tensor negtive options.axes",
+      "inputs": {
+        "input": {
+          "shape": [2, 2, 2, 3],
+          "data": [
+            -1.913036787607325,
+            95.38300711992477,
+            -82.72358015843713,
+            -32.94943032326678,
+            91.42181191536278,
+            -75.1219094678544,
+            -70.81411336302938,
+            -64.39120554424065,
+            -46.982829731397715,
+            91.99561137130661,
+            26.951267233577482,
+            -14.163154887017143,
+            48.66138565461631,
+            -72.48218202176906,
+            -24.52633400273743,
+            84.66979710011137,
+            -27.139524955053318,
+            5.39985222686434,
+            48.46895687574735,
+            2.588408784898675,
+            15.988198777438896,
+            -16.514456459968002,
+            -3.3605690022427694,
+            -88.7479514780816
+          ],
+          "type": "float32"
+        }
+      },
+      "options": {
+        "axes": [-4, -2]
```

Comment:
It's sad imo that negative axes were kept as a policy in WebNN, as these little details complicate WebNN implementations and testing, and they are unnecessary since WebNN doesn't support arbitrary shape inference anyway. Such things could be easily resolved to actual indices by the higher level frameworks. Oh well.